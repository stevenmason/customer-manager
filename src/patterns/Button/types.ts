interface Props extends React.ButtonHTMLAttributes<HTMLButtonElement> {
  href?: string;
}

export default Props;
