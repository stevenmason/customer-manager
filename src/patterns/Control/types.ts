import { SIZES } from "./constants";

export interface Props extends ContainerProps {
  label?: string;
  children: React.ReactNode;
}

export interface ContainerProps {
  size?: SIZES;
}
