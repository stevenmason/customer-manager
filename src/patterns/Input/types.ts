import { CONTROL_SIZES } from "../Control";

export interface Props
  extends Omit<React.InputHTMLAttributes<HTMLInputElement>, "onChange"> {
  label: string;
  controlSize?: CONTROL_SIZES;
  onChange: (value: string, name: string) => void;
}
