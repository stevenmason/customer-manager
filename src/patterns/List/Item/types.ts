interface Props {
  children: React.ReactNode;
}

export default Props;
