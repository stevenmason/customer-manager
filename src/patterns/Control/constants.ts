enum SIZES {
  LARGE = "LARGE",
  SMALL = "SMALL"
}

export { SIZES };
