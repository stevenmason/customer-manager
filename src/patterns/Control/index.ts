import Control from "./Control";
import { SIZES } from "./constants";
export default Control;
export { SIZES as CONTROL_SIZES };
