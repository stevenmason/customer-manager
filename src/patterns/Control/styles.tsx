import styled from "styled-components";
import { SIZES } from "./constants";
import { ContainerProps } from "./types";

const Container = styled.div`
  position: relative;
  display: inline-block;
  color: #5d6369;
  margin: 10px;
  min-width: ${(props: ContainerProps) =>
    props.size === SIZES.LARGE ? 200 : 150}px;
`;

const Label = styled.label`
  display: inline-block;
  margin-bottom: 0.5rem;
  font-weight: bold;
  color: #5a6066;
`;

export { Container, Label };
