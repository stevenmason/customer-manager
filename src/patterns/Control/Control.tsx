import React from "react";

import { SIZES } from "./constants";
import { Props } from "./types";
import * as SC from "./styles";

const Control: React.FC<Props> = ({ children, label, size = SIZES.LARGE }) => (
  <SC.Container size={size}>
    {label && <SC.Label>{label}</SC.Label>}
    {children}
  </SC.Container>
);

export default Control;
