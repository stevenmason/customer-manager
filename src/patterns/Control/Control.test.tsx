import React from "react";
import renderer from "react-test-renderer";

import Control from "./Control";
import { Props } from "./types";
import { SIZES } from "./constants";

const PROPS: Props = {
  label: "Label",
  size: SIZES.LARGE,
  children: "hi there"
};

it("renders correctly", () => {
  const tree = renderer.create(<Control {...PROPS} />).toJSON();
  expect(tree).toMatchSnapshot();
});
