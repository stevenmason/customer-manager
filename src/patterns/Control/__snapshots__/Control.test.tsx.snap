// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-bdVaJa bUbkJD"
  size="LARGE"
>
  <label
    className="sc-bwzfXH jNTJqm"
  >
    Label
  </label>
  hi there
</div>
`;
